### ISSUE TITLE
TransformationsFilter does not transform component position when slanting

### ISSUE THREAD

**Issue by khaledhosny** [googlefonts/ufo2ft#378]
I’m using `TransformationsFilter` to slant a font, but almost all my components are displaced after slanting.

This seems to be caused by:
https://github.com/googlefonts/ufo2ft/blob/66510c8128f9a447617c8c4d6d5871ef4577f74f/Lib/ufo2ft/filters/transformations.py#L30-L34

But I think components will at least need their `x` and `y` positions transformed or they will be displaced.

**Comment by khaledhosny** [comment]
Check [Amiri-Regular.ufo.zip](https://github.com/googlefonts/ufo2ft/files/4647374/Amiri-Regular.ufo.zip) and the following script:
```python
from ufo2ft.filters.transformations import TransformationsFilter
from ufoLib2 import Font

font = Font("Amiri-Regular.ufo")
slant = TransformationsFilter(Slant=-10)
slant(font)
font.save("Amiri-Slanted.ufo")
```

Check for example `uni0773` glyphs in the slanted font (above is ufo2ft result, below is the expected result):
<img width="307" alt="image" src="https://user-images.githubusercontent.com/93914/82268303-fb500880-996e-11ea-9499-f4c4d23764c5.png">

**Comment by khaledhosny** [comment]
My analysis might not be correct, but I still believe it is a bug.

**Comment by anthrotype** [comment]
hm yeah, looks wrong. I believe that whole if block should be removed. It also looks like the tests are not covering those lines.

Can you send a PR? If not, I can do this some time next week.

**Comment by anthrotype** [comment]
this is somewhat similar to the issue I encountered in https://github.com/robotools/fontMath/pull/194

in general, we want to avoid applying the transformation twice to base glyphs as well as their components. Here it's complicated by the fact that filters can selectively include/exclude glyphs, so depending on whether both a base glyph and the composites that refer to it have been included, or only one of the kind, we should do the right thing. It's been long time since I wrote this and can't fully remember all details.
